### PR TITLE
[UWP] Thumb color and image for Slider

### LINKED ISSUE #1684: [Enhancement]  Color on Slider
## Rationale

Slider doesn't offer much customization currently. Add some basic cross platform properties to reduce need for custom renderers

## Implementation

Add ThumbColor, Color, and possibly ThumbImage properties 

```csharp

public static readonly BindableProperty ColorProperty = BindableProperty.Create(nameof(ThumbColor), typeof(Color), typeof(Slider), default(Color));

public static readonly BindableProperty ThumbColorProperty = BindableProperty.Create(nameof(ThumbColor), typeof(Color), typeof(Slider), default(Color));

public static readonly BindableProperty ThumbImageProperty = BindableProperty.Create(nameof(ThumbImage), typeof(FileImageSource), typeof(Slider), default(FileImageSource));
```

### Android
When the Color property of the slider is changed this will be reflected on the slide bar.
When the ThumbColor property is changed the Thumb part of the slider will change
If the user specifies an image for the Thumb then an image will be used instead

ThumbColor and ThumbImage are mutually exclusive

### iOS
See Android

### UWP
See Android

## Implications for CSS

```css
slider {color: Red }

//with possible addition of pseudo elements
slider::thumb {color: Red }
slider::thumb {background-image: path }
```

## Backward Compatibility
Third party renderers may need to be updated to ensure that this functionality is supported through the new official mechanism. Further we will need to be careful to code the changes to the renderers in a careful manner to ensure that if someone is already using an effect support this feature that the effect is as best as possible not broken by our changes.  The current default color palette should be maintained so that after this is implemented everyone's current sliders don't change

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

This is the UWP implementation of thumb image/color (fixes #1684). It is an addendum to the changes in PR #1781.

This also adds the new Slider properties to the Slider Gallery.

### CODE REVIEW

**Comment by jassmith** [comment]
I hate how you cant easily see what you changed from the default style

File: Xamarin.Forms.Platform.UAP/SliderStyle.xaml [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                  Background="{ThemeResource SystemControlForegroundAccentBrush}"
+                  Style="{StaticResource SliderThumbStyle}"
+                  DataContext="{TemplateBinding Value}"
+                  Height="24"
+                  Width="8"
+                  Grid.Row="0"
+                  Grid.RowSpan="3"
+                  Grid.Column="1"
+                  AutomationProperties.AccessibilityView="Raw" />
+								<Thumb x:Name="HorizontalImageThumb"
```
Comment (hartez) [inline Xamarin.Forms.Platform.UAP/SliderStyle.xaml]: @jassmith IIRC, the only change from the default style was adding the HorizontalImageThumb.

File: Xamarin.Forms.Platform.UAP/SliderStyle.xaml [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+										<ControlTemplate TargetType="Thumb">
+											<Border Background="{TemplateBinding Background}"
+                      BorderBrush="{TemplateBinding BorderBrush}"
+                      BorderThickness="{TemplateBinding BorderThickness}"
+                      CornerRadius="4" />
+										</ControlTemplate>
+									</Setter.Value>
+								</Setter>
+							</Style>
+							<Style TargetType="Thumb" x:Key="SliderThumbImageStyle">
```
Comment (hartez) [inline Xamarin.Forms.Platform.UAP/SliderStyle.xaml]: @jassmith And the SliderThumbImageStyle.

File: Xamarin.Forms.Platform.UAP/FormsSlider.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Xamarin.Forms.Platform.UWP
+{
+	public class FormsSlider : Windows.UI.Xaml.Controls.Slider
+	{
+		internal Thumb Thumb { get; set; }
+		internal Thumb ImageThumb { get; set; }
+
+		public static readonly DependencyProperty ThumbImageProperty = DependencyProperty.Register("ThumbImage", typeof(BitmapImage),
```
Comment (samhouts) [inline Xamarin.Forms.Platform.UAP/FormsSlider.cs]: `nameof(ThumbImage)`?

Comment (hartez) [reply Xamarin.Forms.Platform.UAP/FormsSlider.cs]: Fixed.

**Pull request merged**